core/document/src/document.rs

```diff
@@ -354,6 +354,14 @@ impl Document {
 				let path = path.as_slice()[..path.len() - 1].to_vec();
 				Some(vec![DocumentResponse::DocumentChanged, DocumentResponse::FolderChanged { path }])
 			}
+			Operation::FillLayer { path, color } => {
+				if let Ok(layer) = self.layer_mut(path) {
+					layer.style.set_fill(Some(layers::style::Fill::new(*color)));
+					layer.cache_dirty = true;
+				}
+				self.root.cache_dirty = true;
```

Comment:
Could you while your at it maybe include this snippet of code?
```rust
fn mark_as_dirty(&mut self, path: &[PathId]) {
   let mut root = self.root
   for id in path {
       root.dirty = true;
       root = root.as_folder().layer(id) // skip in last iteration
  }
}
```
That probably won't compile but that should not be to hard to fix if you look at the other functions, would just be nice to do the dirtification properly

Reply:
I was able to get that function compiled, but I couldn't get it to actually properly dirty the cache. I tried some variations with no luck; have you been able to get this function working?

Reply:
did you try some debug printing? I would imagine it to be relatively easy to debug

Reply:
Done, I had to change the order of the `root.dirty = true` statements to ensure that the last layer in the path was dirtied.